app/src/main/java/com/varqulabs/dolarblue/history/presentation/HistoryViewModel.kt

```diff
@@ -2,23 +2,317 @@ package com.varqulabs.dolarblue.history.presentation
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.varqulabs.dolarblue.core.domain.DataState
 import com.varqulabs.dolarblue.core.presentation.utils.mvi.MVIContract
 import com.varqulabs.dolarblue.core.presentation.utils.mvi.mviDelegate
-import com.varqulabs.dolarblue.history.presentation.HistoryEvent.OnClickDrawer
+import com.varqulabs.dolarblue.history.domain.model.Conversion
+import com.varqulabs.dolarblue.history.domain.model.QueryAndCurrency
+import com.varqulabs.dolarblue.history.domain.useCases.DeleteConversionUseCase
+import com.varqulabs.dolarblue.history.domain.useCases.DeleteExchangeRateUseCase
+import com.varqulabs.dolarblue.history.domain.useCases.UpdateConversionUseCase
+import com.varqulabs.dolarblue.history.domain.useCases.GetConversionsHistoryFlowUseCase
+import com.varqulabs.dolarblue.history.domain.useCases.GetFavoriteConversionsHistoryUseCase
+import com.varqulabs.dolarblue.history.domain.useCases.GetExchangeRateConversionCountUseCase
+import com.varqulabs.dolarblue.history.domain.useCases.SearchConversionsHistoryUseCase
+import com.varqulabs.dolarblue.history.presentation.HistoryEvent.*
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class HistoryViewModel @Inject constructor(
-
-) : ViewModel(), MVIContract<HistoryState, HistoryEvent, HistoryUiEffect> by mviDelegate(HistoryState()) {
+    private val getConversionsHistoryUseCase: GetConversionsHistoryFlowUseCase,
+    private val searchConversionsHistoryUseCase: SearchConversionsHistoryUseCase,
+    private val updateConversionUseCase: UpdateConversionUseCase,
+    private val getFavoriteConversionsHistoryUseCase: GetFavoriteConversionsHistoryUseCase,
+    private val deleteConversionUseCase: DeleteConversionUseCase,
+    private val getExchangeRateConversionCountUseCase: GetExchangeRateConversionCountUseCase,
+    private val deleteExchangeRateUseCase: DeleteExchangeRateUseCase
+) : ViewModel(),
+    MVIContract<HistoryState, HistoryEvent, HistoryUiEffect> by mviDelegate(HistoryState()) {
 
     override fun eventHandler(event: HistoryEvent) {
         when (event) {
             is OnClickDrawer -> emitOpenDrawer()
```

Comment:
el eventHandler queda un poco extenso con logica en los {} de cada evento

Separemos en funciones aparte, buscando mejorar legibilidad

Reply:
En lugar de esto
is OnClickGetFavoriteConversions -> {
                updateUi { copy(showFavoriteConversions = !showFavoriteConversions) }
                showConversions(uiState.value.showFavoriteConversions)
            }
            
Algo así?            
is OnClickGetFavoriteConversions -> function()

Reply:
Si